### ISSUE TITLE
Spectra Client Doesn't Launch After Restart

### ISSUE THREAD

**Issue by Nekonyx** [ValoSpectra/Spectra-Client#8]
Spectra Client fails to start after being closed and then launched again. I don't have exact reproduction steps, but the issue is most frequently encountered under the following conditions:  

1. Launch Spectra Client.  
2. Close Spectra Client.  
3. Attempt to launch it again.  

**Expected Behavior:**  
Spectra Client should start successfully.  

**Actual Behavior:**  
Spectra Client does not start.  

**Steps to Reproduce:**  
1. Launch Spectra Client.  
2. Close the application.  
3. Try launching it again.  

**Additional Information:**  
When launching Spectra Client via the command line, the following output is displayed:  
```
00:34:38.170 > [Updater]  CHANNEL =  latest
00:34:38.172 > [Updater] App path =  C:\Users\foo\AppData\Local\Programs\Spectra Client
00:34:38.174 > [Updater] Booting
00:34:39.140 > [Updater] Checking for updates...
00:34:39.141 > Checking for update
00:34:39.142 > [Updater] Attaching listeners
00:34:39.143 > [Updater] Added on quit listener
00:34:41.169 > Update for version 0.2.7 is not available (latest version: 0.2.7, downgrade is disallowed).
00:34:41.169 > [Updater] Update not available
00:34:41.170 > [Updater] Booted
00:34:41.171 > GEP version 267.0.4 ready!
(node:28832) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'setTitle')
    at C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\services\gepService.js:53:22
    at node:electron/js2c/browser_init:2:43583
    at Array.forEach (<anonymous>)
    at OverwolfPackageApi.invokeEventToListenerIfNeeded (node:electron/js2c/browser_init:2:43570)
    at OverwolfPackageApi.on (node:electron/js2c/browser_init:2:43404)
    at GameEventsService.registerOverwolfPackageManager (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\services\gepService.js:48:31)
    at new GameEventsService (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\services\gepService.js:33:14)
    at C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:72:18
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:5:58)
(Use `Spectra Client --trace-warnings ...` to show where the warning was created)
(node:28832) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:28832) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'info')
    at App.onQuit (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\node_modules\@electron-delta\updater\src\index.js:293:17)
    at App.emit (node:events:531:35)
    at App.<anonymous> (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:85:13)
    at App.emit (node:events:531:35)
(node:28832) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
```

**Comment by Nekonyx** [comment]
Output with `--trace-warnings`:
```
00:44:37.249 > [Updater]  CHANNEL =  latest
00:44:37.251 > [Updater] App path =  C:\Users\foo\AppData\Local\Programs\Spectra Client
00:44:37.253 > [Updater] Booting
00:44:37.937 > [Updater] Checking for updates...
00:44:37.937 > Checking for update
00:44:37.938 > [Updater] Attaching listeners
00:44:37.939 > [Updater] Added on quit listener
00:44:40.854 > Update for version 0.2.7 is not available (latest version: 0.2.7, downgrade is disallowed).
00:44:40.855 > [Updater] Update not available
00:44:40.856 > [Updater] Booted
00:44:40.857 > GEP version 267.0.4 ready!
(node:123264) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'setTitle')
    at C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\services\gepService.js:53:22
    at node:electron/js2c/browser_init:2:43583
    at Array.forEach (<anonymous>)
    at OverwolfPackageApi.invokeEventToListenerIfNeeded (node:electron/js2c/browser_init:2:43570)
    at OverwolfPackageApi.on (node:electron/js2c/browser_init:2:43404)
    at GameEventsService.registerOverwolfPackageManager (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\services\gepService.js:48:31)
    at new GameEventsService (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\services\gepService.js:33:14)
    at C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:72:18
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:5:58)
    at emitUnhandledRejectionWarning (node:internal/process/promises:298:15)
    at warnWithErrorCodeUnhandledRejectionsMode (node:internal/process/promises:406:5)
    at processPromiseRejections (node:internal/process/promises:470:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)
(node:123264) TypeError: Cannot read properties of undefined (reading 'setTitle')
    at C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\services\gepService.js:53:22
    at node:electron/js2c/browser_init:2:43583
    at Array.forEach (<anonymous>)
    at OverwolfPackageApi.invokeEventToListenerIfNeeded (node:electron/js2c/browser_init:2:43570)
    at OverwolfPackageApi.on (node:electron/js2c/browser_init:2:43404)
    at GameEventsService.registerOverwolfPackageManager (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\services\gepService.js:48:31)
    at new GameEventsService (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\services\gepService.js:33:14)
    at C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:72:18
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:5:58)
(node:123264) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'info')
    at App.onQuit (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\node_modules\@electron-delta\updater\src\index.js:293:17)
    at App.emit (node:events:531:35)
    at App.<anonymous> (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:85:13)
    at App.emit (node:events:531:35)
    at emitUnhandledRejectionWarning (node:internal/process/promises:298:15)
    at warnWithErrorCodeUnhandledRejectionsMode (node:internal/process/promises:406:5)
    at processPromiseRejections (node:internal/process/promises:470:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)
(node:123264) TypeError: Cannot read properties of undefined (reading 'info')
    at App.onQuit (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\node_modules\@electron-delta\updater\src\index.js:293:17)
    at App.emit (node:events:531:35)
    at App.<anonymous> (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:85:13)
    at App.emit (node:events:531:35)
```

**Comment by DarkView** [comment]
> **Proposed Solution:** I cloned the repository locally and removed the following lines:
> 
> https://github.com/ValoSpectra/Spectra-Client/blob/c7bc4d01d3a9ce1679a2694d134e4ac62b7a0ee8/src/main.ts#L53-L63
> 
> After making this change, everything worked perfectly.  

Removing the updater is not a good solution to this issue.
Please try uninstalling the Client and re-downloading the latest version of the client from [here](https://github.com/ValoSpectra/Spectra-Client/releases/latest) (Spectra-Client-Setup-0.2.7.exe).
Please use the uninstaller present in the files.
Let us know if the issue persists after that.

**Comment by Nekonyx** [comment]
> > **Proposed Solution:** I cloned the repository locally and removed the following lines:
> > https://github.com/ValoSpectra/Spectra-Client/blob/c7bc4d01d3a9ce1679a2694d134e4ac62b7a0ee8/src/main.ts#L53-L63
> > 
> > After making this change, everything worked perfectly.
> 
> Removing the updater is not a good solution to this issue. Please try uninstalling the Client and re-downloading the latest version of the client from [here](https://github.com/ValoSpectra/Spectra-Client/releases/latest) (Spectra-Client-Setup-0.2.7.exe). Please use the uninstaller present in the files. Let us know if the issue persists after that.

I deleted my "proposed solution" because it hasn't the case. It turns out that cloning the repository locally and running the application using `npm run start` works perfectly every time. And it doesn't matter whether I'm removing those lines or not.

**Comment by Nekonyx** [comment]
> Let us know if the issue persists after that.

Reinstallation worked for me. However, this problem wasn't unique to me.
Few hours ago, I asked another person, who had never used Spectra Client before, to download it (the latest version) and run it. They were able to install it, launch it, and observe one match, but after restarting the program, it refused to launch a second time.

**Comment by Nekonyx** [comment]
Update: Spectra Client has stopped launching for me again.

**Comment by DarkView** [comment]
Does this issue persist through a windows restart?

**Comment by Nekonyx** [comment]
> Does this issue persist through a windows restart?


All I can say is that after restarting, Spectra Client launches, but after 2-3 closures, it stops launching. All I see is the "Checking for updates" window, and then nothing happens.

**Comment by Nekonyx** [comment]
I'm not sure if it's normal for it to log these errors to the console after the program is gracefully closed, but after these lines, Spectra Client definitely won't launch anymore:
```
01:08:14.700 > [Updater]  CHANNEL =  latest
01:08:14.702 > [Updater] App path =  C:\Users\foo\AppData\Local\Programs\Spectra Client
01:08:14.705 > [Updater] Booting
01:08:15.395 > [Updater] Checking for updates...
01:08:15.396 > Checking for update
01:08:15.397 > [Updater] Attaching listeners
01:08:15.397 > [Updater] Added on quit listener
01:08:17.463 > Update for version 0.2.7 is not available (latest version: 0.2.7, downgrade is disallowed).
01:08:17.464 > [Updater] Update not available
01:08:17.465 > [Updater] Booted
01:08:17.505 > Starting Overwolf Setup
01:08:17.668 > GEP version 267.0.4 ready!
(node:23272) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'info')
    at App.onQuit (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\node_modules\@electron-delta\updater\src\index.js:293:17)
    at App.emit (node:events:531:35)
    at App.<anonymous> (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:85:13)
    at App.emit (node:events:531:35)
    at emitUnhandledRejectionWarning (node:internal/process/promises:298:15)
    at warnWithErrorCodeUnhandledRejectionsMode (node:internal/process/promises:406:5)
    at processPromiseRejections (node:internal/process/promises:470:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)
(node:23272) TypeError: Cannot read properties of undefined (reading 'info')
    at App.onQuit (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\node_modules\@electron-delta\updater\src\index.js:293:17)
    at App.emit (node:events:531:35)
    at App.<anonymous> (C:\Users\foo\AppData\Local\Programs\Spectra Client\resources\app.asar\app\main.js:85:13)
    at App.emit (node:events:531:35)
```

**Comment by Tiranthine** [comment]
So I took a look at the code and I think I found the issue.
It should be fixed with 7a32c87. We've bumped the client version to 0.2.8.
Could you see if you still have the issue with the new client version.

**Comment by Nekonyx** [comment]
> So I took a look at the code and I think I found the issue. It should be fixed with [7a32c87](https://github.com/ValoSpectra/Spectra-Client/commit/7a32c87297a14ae7119ebe3065ab394a031b0c2f). We've bumped the client version to 0.2.8. Could you see if you still have the issue with the new client version.

It seems like the issue is resolved. Dozens of restarts, and everything works as it should!
